File: lan_folder_sync/file_sender.py
import socket
import sys
from framing import send_file

HOST = sys.argv[1] if len(sys.argv) > 1 else "127.0.0.1"
PORT = 8765
path = sys.argv[2] if len(sys.argv) > 2 else "bigfile.bin"

sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
sock.connect((HOST, PORT))
send_file(sock, ".", path)
print(f"Sent {path}")
sock.close()

File: lan_folder_sync_swift/Sources/file_sender/FileSender.swift
// The first experiment: push one file at file_receiver. No HELLO, no manifest.
import SyncCore

@main
struct FileSender {
    static func main() {
        setUpProcess()
        let args = CommandLine.arguments
        let host = args.count > 1 ? args[1] : "127.0.0.1"
        let path = args.count > 2 ? args[2] : "bigfile.bin"

        do {
            let sock = try Socket.connect(host: host, port: 8765, timeout: 8)
            if try sendFile(sock, root: ".", relPath: path) {
                print("  ! \(path) changed while being sent; the copy is padded or clipped")
            }
            print("Sent \(path)")
            sock.close()
        } catch {
            die("send failed: \(error)")
        }
    }
}
